src/python/pants/report.py

```diff
@@ -0,0 +1,20 @@
+import uuid
+
+from humbug.consent import environment_variable_opt_out, HumbugConsent, no
+from humbug.report import Reporter
+
+BUGOUT_ACCESS_TOKEN: str = "3ae76900-9a68-4a87-a127-7c9f179d7272"
+BUGOUT_JOURNAL_ID: str = "801e9b3c-6b03-40a7-870f-5b25d326da66"
+
+# To opt out of reporting, set PANTS_REPORTING_ENABLED=0 in your environment.
+reporting_consent: HumbugConsent = HumbugConsent(environment_variable_opt_out("PANTS_REPORTING_ENABLED", no))
```

Comment:
We would probably want to tie this into our existing options system.

Reply:
Yeah, I think we do want to tie it into our options system. I'd like for users to be able to set opt in/out in `pants.toml`, rather than env var. It would be neat if we could do something like `[stats].share`. 

But @stuhood I think it may be non-trivial to access that subsystem in the run tracker. Maybe this should use a streaming workunit handler that only runs when `finished=True`? If this is too hard, we could use a global option.

Reply:
I think it's reasonable for opt-in/opt-out of reporting to be a global option in any case, since it's something that affects pants globally.

Reply:
Here is an example of a consent mechanism that reads consent from a file:
```
https://github.com/activeloopai/Hub/pull/624/files#diff-77c3972b46d0968e0c1b74539da30b992c9a2a71a08bd7ebc6932d400e15a090R22
```

Will look into putting the consent in `pants.toml`.

Reply:
I wouldn't worry about that - it requires registering a new option, which we can handle on our end. This is fine as-is for now, as a guideline.

Reply:
@Eric-Arellano : The `RunTracker` is constructed with the complete `Options` (which is set as `self._all_options`), so it should be fine to read an option value.